### PR TITLE
Add `up` alias for build, push, deploy

### PR DESCRIPTION
**Pull request by johnmccabe**
## Description
Add `up` alias for `build` ➡︎ `push` ➡︎ `deploy`.

faas-up

faas-up-skip-push

~~This reuses all the flags/logic from the respective commands which are chained together, it adds a single local flag `--push` which is used to include the push step.~~

~~Alternatives to setting `--push` would be to do so for non-`localhost`/`127.0.0.1` gateways perhaps. @alexellis wdyt?~~

This reuses all the flags/logic from the respective commands which are chained together, it adds a single local flag `--skip-push` which is used to exclude the push step.

## Motivation and Context

- [x] I have raised an issue to propose this change (#472)

## How Has This Been Tested?

Built and tested against a local gateway with both a local and docker hub image push/skip-push.

No unit tests added as it just reuses the underlying commands.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I've read the [CONTRIBUTION](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md) guide
- [x] I have signed-off my commits with `git commit -s`
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by alexellis** [comment]
Let's just have `faas-cli push` refuse if there is no prefix on the image. I think that's a separate change though. cc @viveksyngh

File: commands/up.go [hunk 1]
```diff
@@ -0,0 +1,69 @@
+// Copyright (c) Alex Ellis 2017. All rights reserved.
```
Comment (viveksyngh) [inline commands/up.go]: Please use `OpenFaaS Author(s) 2018` for copyright. One of the feedback from @alexellis  on my previous PR :)

Comment (alexellis) [reply commands/up.go]: +1 for this

Comment (johnmccabe) [reply commands/up.go]: yup, its a WIP pr was going to address this later - funnily enough I was looking for that header text when raising the PR and noticed this - https://github.com/openfaas/faas-cli/blob/master/commands/update_gitignore.go, you'll need to update that one as well :)

**Comment by burtonr** [comment]
Awesome @ivanayov  I was wondering the same thing, but haven't had time to test it out yet. Good to know!

**Pull request merged**